AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/OrganizationControllerTests.cs

```diff
@@ -28,246 +30,291 @@ public class OrganizationControllerTests
+        {
+            var mediator = new Mock<IMediator>();
+            mediator.Setup(x => x.SendAsync(It.IsAny<OrganizationEditQuery>())).ReturnsAsync(null);
+
+            var sut = new OrganizationController(mediator.Object, null);
+            var result = await sut.Edit(It.IsAny<int>());
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async Task EditGetReturnsCorrectViewAndViewModel_WhenMediatorReturnsOrganizationEditModel()
+        {
+            CreateSut();
+
+            var model = new OrganizationEditViewModel();
+
+            _mediator.Setup(x => x.SendAsync(It.Is<OrganizationEditQuery>(y => y.Id == Id))).ReturnsAsync(model);
+
+            var result = (ViewResult)await _sut.Edit(Id);
+
+            Assert.Equal("Edit", result.ViewName);
+            Assert.Same(model, result.ViewData.Model);
+        }
+
+        [Fact]
+        public async Task EditPostReturnsBadRequestResult_WhenModelIsNull()
+        {
+            var sut = new OrganizationController(null, null);
+            var result = await sut.Edit(null);
+            Assert.IsType<BadRequestResult>(result);
+        }
+
+        [Fact]
+        public async Task EditPostInvokesOrganizationValidatorsValidateMethodWithTheCorrectOrganizationEditModel_WhenModelIsNotNull()
+        {
+            //Arrange
+            var organizationValidatorMock = new Mock<IOrganizationEditModelValidator>();
+            var errors = new List<KeyValuePair<string, string>> { new KeyValuePair<string, string>("FakeProperty", "FakeError") };
+            organizationValidatorMock.Setup(o => o.Validate(It.IsAny<OrganizationEditViewModel>())).Returns(errors);
+            var sut = new OrganizationController(null, organizationValidatorMock.Object);
+
+            //Act
+            var result = await sut.Edit(_organizationEditModel) as ViewResult;
+
+            //Assert
+            organizationValidatorMock.Verify(o => o.Validate(It.IsAny<OrganizationEditViewModel>()), Times.Once);
+            result.ShouldNotBeNull();
+            result.ViewData.Model.ShouldBeSameAs(_organizationEditModel);
+        }
+
+        [Fact]
+        public async Task EditPostAddsTheCorrectErrorsToModelState_WhenOrganizationValidatorHasErrors_AndModelIsNotNull()
+        {
+            //Arrange
+            var organizationValidatorMock = new Mock<IOrganizationEditModelValidator>();
+            var errors = new List<KeyValuePair<string, string>> { new KeyValuePair<string, string>("FakeProperty", "FakeError") };
+            organizationValidatorMock.Setup(o => o.Validate(It.IsAny<OrganizationEditViewModel>())).Returns(errors);
+            var sut = new OrganizationController(null, organizationValidatorMock.Object);
+
+            //Act
+            var result = await sut.Edit(_organizationEditModel) as ViewResult;
+
+            //Assert
+            result.ViewData.Model.ShouldBeSameAs(_organizationEditModel);

```

Comment:
you can remove this assertion, as this unit test has nothing to do with asserting the correct model is being returned, but is asserting that the correct error was added to model state, which you wrote perfectly.